backend/module/bot/slackbot/slackbot.go

```diff
@@ -0,0 +1,168 @@
+package slackbot
+
+// <!-- START clutchdoc -->
+// description: Receives bot events from the Slack Events API
+// <!-- END clutchdoc -->
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/slack-go/slack"
+	"github.com/slack-go/slack/slackevents"
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	slackbotv1 "github.com/lyft/clutch/backend/api/bot/slackbot/v1"
+	slackbotconfigv1 "github.com/lyft/clutch/backend/api/config/module/bot/slackbot/v1"
+	"github.com/lyft/clutch/backend/gateway/log"
+	"github.com/lyft/clutch/backend/module"
+	"github.com/lyft/clutch/backend/service"
+	"github.com/lyft/clutch/backend/service/bot"
+)
+
+const (
+	Name      = "clutch.module.bot.slackbot"
+	dmMessage = "im"
+)
+
+func New(cfg *anypb.Any, logger *zap.Logger, scope tally.Scope) (module.Module, error) {
+	config := &slackbotconfigv1.Config{}
+
+	if err := cfg.UnmarshalTo(config); err != nil {
+		return nil, err
+	}
+
+	svc, ok := service.Registry["clutch.service.bot"]
+	if !ok {
+		return nil, errors.New("could not find service")
+	}
+
+	bot, ok := svc.(bot.Service)
+	if !ok {
+		return nil, errors.New("service was not the correct type")
+	}
+
+	m := &mod{
+		slack:             slack.New(config.BotToken),
+		verificationToken: config.VerificationToken,
+		bot:               bot,
+		logger:            logger,
+		scope:             scope,
+	}
+
+	return m, nil
+}
+
+type mod struct {
+	slack             *slack.Client
+	verificationToken string
+	bot               bot.Service
+	logger            *zap.Logger
+	scope             tally.Scope
+}
+
+func (m *mod) Register(r module.Registrar) error {
+	slackbotv1.RegisterSlackBotAPIServer(r.GRPCServer(), m)
+	return r.RegisterJSONGateway(slackbotv1.RegisterSlackBotAPIHandler)
+}
+
+// One request, one event. https://api.slack.com/apis/connections/events-api#the-events-api__receiving-events
+// TODO: (sperry) capture success/failure metrics when handling the event
+func (m *mod) Event(ctx context.Context, req *slackbotv1.EventRequest) (*slackbotv1.EventResponse, error) {
+	err := verifySlackRequest(m.verificationToken, req.Token)
+	if err != nil {
+		m.logger.Error("verify slack request error", log.ErrorField(err))
+		return nil, status.Error(codes.Unauthenticated, err.Error())
+	}
+
+	m.logger.Info("received event from Slack API")
```

Comment:
delete or add more info

Reply:
realized the placement of this log at this step of the flow isnt useful b/c we're not yet sure what kind of an event it is. moved it